torch/csrc/jit/passes/replacement_of_old_operators.cpp

```diff
@@ -28,20 +28,28 @@ struct OldOpsReplacerWithUpgraders {
     DepthFirstGraphNodeIterator graph_it(graph_);
     Node* node = graph_it.next();
     while (node) {
-      if (auto schema = node->maybeSchema()) {
-        auto schema_name = getFullSchemaName(*schema);
+      // load the schema name for this op
+      c10::optional<std::string> schema_name = c10::nullopt;
+      if (auto op_schema = node->maybeSchema()) {
+        schema_name = getFullSchemaName(*op_schema);
+      } else {
+        schema_name = node->getHistoricSchemaName();
+      }
+
+      if (schema_name.has_value()) {
```

Comment:
Shouldn't it error out when this condition is not met? In fact, the logic would be clearer and less error prone if we check for `!schema_name.has_value()` and error out when true.

Reply:
I think it shouldn't error out because nodes don't need to have schemas